pynumdiff/total_variation_regularization/_total_variation_regularization.py

```diff
@@ -253,10 +253,13 @@ def jerk_sliding(x, dt, params=None, options=None, gamma=None, solver=None, wind
     elif gamma == None:
         raise ValueError("`gamma` must be given.")
 
-    if len(x) < window_size:
-        warn("len(x) <= window_size, calling standard jerk() without sliding")
+    if len(x) < window_size or window_size < 15:
+        warn("len(x) should be > window_size >= 15, calling standard jerk() without sliding")
         return _total_variation_regularized_derivative(x, dt, 3, gamma, solver=solver)
 
+    if window_size % 2 == 0:
+        window_size += 1 # has to be odd
+        warn("Kernel window size should be odd. Added 1 to length.")
     ramp = window_size//5
```

Comment:
If the `window_size` is too small, this can end up being 0. If I set it to have a minimum value of 1, then that change alone is insufficient and causes indexing errors deeper in the `slide_function` or divide-by-zero errors even deeper in the solvers. Perhaps there is a way to catch and deal with those, but not without more significant debugging and tracing. Instead, I've decided sliding jerk isn't really a reasonable thing to do below a certain window size and am simply calling `jerk` in that case.